inventory.js

```diff
@@ -0,0 +1,219 @@
+// ('items'). This allows us to keep the alternating background colors functioning.
+
+function sort(a, b) {
+  const attrA = a.getAttribute('name');
+  const attrB = b.getAttribute('name');
+
+  return (attrA < attrB) ? -1 : (attrA > attrB) ? 1 : 0;
+}
+
+// Passively update the query param value in the url
+function updateUrlQueryParam(queryParamValue) {
+  window.history.pushState({}, '', `inventory.html?cat=${encodeURIComponent(queryParamValue).toLowerCase()}`);
+}
+
+// Find the category-item that corresponds to the current cat query param
+function getQueryParamCat(queryParam) {
+  return $(`.category-item[data-query_name='${queryParam}']`);
+}
+
+// Called when the user clicks on a category.
+function categoryClick(categoryRow) {
+  // Update the query param to match current selection
+  updateUrlQueryParam(categoryRow[0].dataset.query_name);
+
+  // Reset the search field
+  $('#searchfield')[0].value = '';
+
+  // Highlight none but the selected category.
+  $('category').removeClass('selected');
+  categoryRow.addClass('selected');
+
+  // Sort all rows but the headline and move them to the hidden 'allitems' container.
+  const items = $('itemrow').not('.headline').sort(sort);
+  $.each(items, (idx, itm) => { $('allitems').append(itm); });
+
+  // Move all matching rows into the visible 'items' container.
+  $(`[category_id = ${categoryRow.attr('id')}]`).appendTo('items');
+
+  // Update the headline with the selected category.
+  $('itemrow.headline').find('itemname').html(categoryRow.text());
+}
+
+// Called when the user types into the search field.
+// Gets reset when the user clicks on a category.
+function search(input) {
+  const items = $('itemrow').not('.headline').sort(sort);
+  const queryParam = getParameterByName();
+  const queryParamCat = getQueryParamCat(queryParam);
+  input = input.toUpperCase();
+
+  // If the input is cleared, restore the list based on the query param (if there is one)
+  if (input === '') {
+    return queryParam ? categoryClick($(queryParamCat)) : categoryClick($('category:first'));
+  }
+
+  // Sort all rows but the headline and move them to the hidden 'allitems' container.
+  $.each(items, (idx, itm) => { $('allitems').append(itm); });
+
+  // Remove any highlight from the category list.
+  $('category').removeClass('selected');
+
+  // Loop over all products and move the matching ones to the visible 'items' container.
+  $('itemrow').not('.headline').each(function() {
+    const productName = $(this).find('itemname').text().toUpperCase();
+
+    if (productName.toUpperCase().indexOf(input) > -1) {
+      $(this).appendTo('items');
+    }
+  });
+
+  // Update the headline with the current search term.
+  $('itemrow.headline').find('itemname').html(`Search for ${input}`);
+}
+
+// Called when the DOM is ready. Loads all food items from the database.
+function loadFoodItems(queryParam) {
+  $.post('../inventory_get_all_items.php', {}, (data) => {
```

Comment:
Are the files in different folders now? Maybe I misunderstand why you need the ../

Reply:
Ah, that's a typo - initially, I did move the script into a scripts/ directory, but then when I saw how things were arranged on the server, I opted to move the script back out to the parent - oddly, the script still ran ok on my local machine, so I failed to notice. Fixing this now.

Reply:
Browser had cached the app with the older file structure, so I didn't notice that it should've broken...